### ISSUE TITLE
repeated queries reload from cache

### ISSUE THREAD

**Issue by vjcitn** [ccb-hms/ontology-mapper#41]
can the loaded ontology be reused for multiple queries?  each load event takes close to 2 minutes on my laptop
```
>>> tt.cache_exists("EFO")
True
>>> tt.map_terms(["asthma", "acute bronchitis"], "EFO", use_cache=True)
2023-10-26 05:23:41 INFO [text2term.term_collector]: Loading ontology EFO...
2023-10-26 05:25:26 INFO [text2term.term_collector]: ...done (ontology loading time: 102.84s)
2023-10-26 05:25:26 INFO [text2term.t2t]: Loading cached ontology from: cache/EFO/EFO-term-details.pickle
2023-10-26 05:25:29 INFO [text2term.t2t]: Filtered ontology terms to those of type: class
2023-10-26 05:25:29 INFO [text2term.t2t]: Mapping 2 source terms to EFO
2023-10-26 05:25:36 INFO [text2term.t2t]: ...done (mapping time: 7.35s seconds)
                               Source Term ID  ...  Tags
0  http://ccb.hms.harvard.edu/t2t/R3HRYLSWSDf  ...  None
1  http://ccb.hms.harvard.edu/t2t/R3HRYLSWSDf  ...  None
2  http://ccb.hms.harvard.edu/t2t/R3HRYLSWSDf  ...  None
3  http://ccb.hms.harvard.edu/t2t/RKK7t5637Cy  ...  None
4  http://ccb.hms.harvard.edu/t2t/RKK7t5637Cy  ...  None
5  http://ccb.hms.harvard.edu/t2t/RKK7t5637Cy  ...  None

[6 rows x 7 columns]
>>> tt.map_terms(["asthma", "acute bronchitis"], "EFO", use_cache=True)
2023-10-26 05:25:47 INFO [text2term.term_collector]: Loading ontology EFO...
2023-10-26 05:27:25 INFO [text2term.term_collector]: ...done (ontology loading time: 96.66s)
2023-10-26 05:27:25 INFO [text2term.t2t]: Loading cached ontology from: cache/EFO/EFO-term-details.pickle
2023-10-26 05:27:28 INFO [text2term.t2t]: Filtered ontology terms to those of type: class
2023-10-26 05:27:28 INFO [text2term.t2t]: Mapping 2 source terms to EFO
2023-10-26 05:27:31 INFO [text2term.t2t]: ...done (mapping time: 2.97s seconds)
                               Source Term ID  ...  Tags
0  http://ccb.hms.harvard.edu/t2t/RCJvSJhexwv  ...  None
1  http://ccb.hms.harvard.edu/t2t/RCJvSJhexwv  ...  None
2  http://ccb.hms.harvard.edu/t2t/RCJvSJhexwv  ...  None
3  http://ccb.hms.harvard.edu/t2t/RF9pUeb5Kbb  ...  None
4  http://ccb.hms.harvard.edu/t2t/RF9pUeb5Kbb  ...  None
5  http://ccb.hms.harvard.edu/t2t/RF9pUeb5Kbb  ...  None

[6 rows x 7 columns]
```

**Comment by vjcitn** [comment]
For programmatic usage, once an ontology has been loaded, it would be good to be able to pass the reference to text2term methods.  In the command-line design this is not typical because you always start from scratch.  But if we want to capitalize on this system in R we don't want to load the ontology more than once.  I am not clear on how the pickled ontology data in the cache might be efficiently restored.

**Comment by rsgoncalves** [comment]
Thank you for reporting this Vince. I can confirm we were able to replicate and identify the issue. The cache is indeed intended to make these queries faster without having to reload the ontology, but a little bug sneaked in the last revision that makes the tool always reload an ontology even if it was cached. We are working on a fix and will report back.

**Comment by paynejason** [comment]
This bug has been fixed with 4.1.0. Please let us know if there are any other issues going forward.
